packages/twenty-front/src/modules/object-record/record-show/hooks/useRecordShowContainerTabs.ts

```diff
@@ -12,114 +18,303 @@ import {
+          (obj) =>
+            ![
+              CoreObjectNameSingular.Workflow,
+              CoreObjectNameSingular.WorkflowVersion,
+              CoreObjectNameSingular.WorkflowRun,
+            ].includes(obj),
+        ),
+      },
     },
     {
       id: 'tasks',
       title: 'Tasks',
       Icon: IconCheckbox,
-      hide:
-        targetObjectNameSingular === CoreObjectNameSingular.Note ||
-        targetObjectNameSingular === CoreObjectNameSingular.Task ||
-        isWorkflowRelated,
+      cards: [{ type: CardType.TaskCard }],
+      hide: {
+        onMobile: false,
+        onDesktop: false,
+        inRightDrawer: false,
+        requiresFeatures: [],
+        requiredObjectsNotActive: [CoreObjectNameSingular.Task],
+        requiresRelations: ['taskTargets'],
+        ifCurrentObjectIsNotIn: Object.values(CoreObjectNameSingular).filter(
+          (obj) =>
+            ![
+              CoreObjectNameSingular.Note,
+              CoreObjectNameSingular.Task,
+              CoreObjectNameSingular.Workflow,
+              CoreObjectNameSingular.WorkflowVersion,
+              CoreObjectNameSingular.WorkflowRun,
+            ].includes(obj),
+        ),
+      },
     },
     {
       id: 'notes',
       title: 'Notes',
       Icon: IconNotes,
-      hide:
-        targetObjectNameSingular === CoreObjectNameSingular.Note ||
-        targetObjectNameSingular === CoreObjectNameSingular.Task ||
-        isWorkflowRelated,
+      cards: [{ type: CardType.NoteCard }],
+      hide: {
+        onMobile: false,
+        onDesktop: false,
+        inRightDrawer: false,
+        requiresFeatures: [],
+        requiredObjectsNotActive: [CoreObjectNameSingular.Note],
+        requiresRelations: ['noteTargets'],
+        ifCurrentObjectIsNotIn: Object.values(CoreObjectNameSingular).filter(
+          (obj) =>
+            ![
+              CoreObjectNameSingular.Note,
+              CoreObjectNameSingular.Task,
+              CoreObjectNameSingular.Workflow,
+              CoreObjectNameSingular.WorkflowVersion,
+              CoreObjectNameSingular.WorkflowRun,
+            ].includes(obj),
+        ),
+      },
     },
     {
       id: 'files',
       title: 'Files',
       Icon: IconPaperclip,
-      hide: isWorkflowRelated,
+      cards: [{ type: CardType.FileCard }],
+      hide: {
+        onMobile: false,
+        onDesktop: false,
+        inRightDrawer: false,
+        requiresFeatures: [],
+        requiredObjectsNotActive: [CoreObjectNameSingular.Attachment],
+        requiresRelations: ['attachments'],
+        ifCurrentObjectIsNotIn: Object.values(CoreObjectNameSingular).filter(
+          (obj) =>
+            ![
+              CoreObjectNameSingular.Workflow,
+              CoreObjectNameSingular.WorkflowVersion,
+              CoreObjectNameSingular.WorkflowRun,
+            ].includes(obj),
+        ),
+      },
     },
     {
       id: 'emails',
```

Comment:
Let's transform this array to an object and use the ID as the key of the array, this will make it easier to over write layouts. Let's maybe do the same for the Cards. 
(for inheritance then we can quickly overwrite a given tab/card)